tests/e2e/test_deploy_flow.py

```
@@ -163,3 +165,15 @@ def test_debug_wrapper(self):
                     config_value="False")
         self.deploy(no_push=True)
         self.verify_pod_status(expected_status="Failed")
+
+    def test_undeploy_all(self):
+        """test `mlt undeploy --all` command."""
+        self.init()
+        self.build()
+        self.deploy()
+        self.verify_pod_status()
+        self.status()
+        self.deploy()
+        self.verify_pod_status()
+        self.status()
+        self.undeploy(all_jobs=True)
```

I think this is fine :)
There isn't really a way to pass `all_jobs` to the autouse fixture unless you wanna get super fancy with it, so I'd say just have this as it'd be cleaner than the other way (detecting the name of the test and if it says `undeploy_all` then pass `all_jobs=True` to the `self.undeploy()` call there)

---

on second thought, detecting name would be the way to go because of the outstanding bug of undeploying twice and failing... I'll fix that up next and then we can revert the name detection thing, or you can ignore that the tests are failing and keep going on your PR. WDYT?

---

to me, looks like the work around is a bit complicated, I'm thinking about a fix for the teardown itself. cannot it be fixed it just by adding an exception to the 2nd try https://github.com/IntelAI/mlt/blob/ab00e4b1ef00ccbade39503ad4fc3b4f68a66c89/tests/e2e/test_deploy_flow.py#L44 , I will try this out!

---

🤔 I don't think so, because what if the `undeploy` really did fail? We also wanna test the `undeploy` logic :/ We can't just ignore exceptions when `undeploy`ing, unless you wanna ignore for now the specific error that happens because of the existing `undeploy` bug.